### ISSUE TITLE
Geofence notifications do not execute on Android (works perfectly on iOS)

### ISSUE THREAD

**Issue by ClauderGunpowder** [thudugala/Plugin.LocalNotification#335]
**Describe the bug**
Cannot get Geofence notifications to execute on Android.
Notice: The same geofence code executed on iOS works perfectly, both with the app in for- and background!

I suspected that for my Android version I needed to add a channel group and channel, but when I do I get this error message at startup (see below) so therefore these two lines is commented out in the code sample:

JNINativeWrapper.g.cs : 
NotificationChannelGroup doesn't exist

**To Reproduce**
MauiProgram

using Microsoft.Extensions.Logging;
using Plugin.LocalNotification;
using Plugin.LocalNotification.AndroidOption;
using Plugin.LocalNotification.iOSOption;

namespace PowerFences;

public static class MauiProgram
{
    public static MauiApp CreateMauiApp()
    {
        NotificationChannelGroupRequest notificationChannelGroupRequest =
            new Plugin.LocalNotification.AndroidOption.NotificationChannelGroupRequest()
            {
                Group = "PowerFences",
                Name = "PowerFences"
            };
        NotificationChannelRequest notificationChannelRequest =
            new Plugin.LocalNotification.AndroidOption.NotificationChannelRequest()
        {
            Name = "PowerFences",
            Description = "Channel for Android notifications",
            Group = "PowerFences",
            ShowBadge = true,
            EnableSound = true
        };

        var builder = MauiApp.CreateBuilder();
        builder
            .UseMauiApp<App>()
            .UseLocalNotification(config =>
            {
                config.AddiOS(iOS =>
                {
                    iOS.SetPermission(new iOSNotificationPermission
                    {
                        LocationAuthorization = iOSLocationAuthorization.Always
                    });
                });
                config.AddAndroid(android =>
                {
                    //android.AddChannelGroup(notificationChannelGroupRequest);
                    //android.AddChannel(notificationChannelRequest);
                });
            })
            .ConfigureFonts(fonts =>
            {
                fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
                fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
            });

#if DEBUG
        builder.Logging.AddDebug();
#endif

        return builder.Build();
    }
}

**Expected behavior**
I expect a notification to appear whenever I enter or exit a defined geofence area.

**Platform (please complete the following information):**
 - OS: MacBook Pro 2021

**Smartphone (please complete the following information):**
- Device: iOS Emulator
- Version iPhone 14 iOS 16.1
- Device: Android Emulator
- Version Pixel 5 - API 33

**Additional context**
Using Visual Studio 2022 for Mac
.NET 7, MAUI

Required permissions set for Android: AcceptBackgroundLocation, Access CoarseLocation, AccessFineLocation, AccessNetworkState, Internet, LocationHardware, PostNotifications.
info.plist properties set for iOS: Privacy - Location Always and When In Use Usage Description (true), Privacy - Reminders Usage Description (true), Required background modes.

**Comment by DeerSteak** [comment]
I'm not convinced it's a problem with the plugin as much as it is a problem with Android. I went through the trouble of creating my own BroadcastReceiver in Xamarin.Android and registering my geofences manually, then using the plugin to send notifications, and the behavior didn't really change. 

What fixed it for me was doing everything I could to get more timely location updates. For example, after adding activity tracking, then every time I get into or out of the car, an activity monitoring service fires up and gets location, and that tells the device it's in a geofence. Now notifications deliver like they should. Or at least closer to what I expect. 

Android background location was brutally bad. It's improved a lot with Android 13. Combining it with other services like activity tracking helps a ton.

**Comment by claus-lysholm** [comment]
That's very interesting because I have experienced that if I later fire a method getting my current location, a couple of apparently stacked up but invisible notifications suddenly shows up as visible notifications on the phone.

**Comment by DeerSteak** [comment]
Exactly, because the phone finally got the location and delivered the scheduled notifications.

**Comment by claus-lysholm** [comment]
Notification w/Geofence setup: Do you need to create that in a foreground service or do the plugin manage that itself?
